pkg/trace/api/container_linux_test.go

```diff
@@ -220,15 +212,6 @@ func TestGetContainerID(t *testing.T) {
 		assert.Equal(t, containerID, provider.GetContainerID(req.Context(), req.Header))
 	})
 
-	t.Run("No header with a PID", func(t *testing.T) {
-		ctx := context.WithValue(context.Background(), ucredKey{}, &syscall.Ucred{Pid: containerPID})
-		req, err := http.NewRequestWithContext(ctx, "GET", "http://example.com", nil)
-		if !assert.NoError(t, err) {
-			t.Fail()
-		}
-		assert.Equal(t, containerID, provider.GetContainerID(req.Context(), req.Header))
-	})
-
```

Comment:
maybe replace this test with the new logic? What other coverage exists for the change?